python/lsst/pipe/tasks/drpAssociationPipe.py

```diff
@@ -270,7 +270,9 @@ def visitDetectorPair(dataRef):
                                                    innerTractSkyRegion,
                                                    skyInfo.wcs)
                 associatedSsSources = associatedSsSources[ssInTractPatch]
-                associatedSsDiaSources = associatedSsDiaSources[ssInTractPatch]
+                assocDiaSrcIds = set(associatedSsSources['diaSourceId'])
+                diaSrcMask = [ID in assocDiaSrcIds for ID in associatedSsDiaSources['diaSourceId']]
+                associatedSsDiaSources = associatedSsDiaSources[np.array(diaSrcMask)]
```

Comment:
Could this ever return empty? If so, is that okay, or is it handled somewhere else?